Source/WebCore/mathml/MathMLAnnotationElement.cpp

```diff
@@ -71,35 +71,13 @@ bool MathMLAnnotationElement::childShouldCreateRenderer(const Node& child) const
     if (hasTagName(MathMLNames::annotationTag))
         return child.isTextNode();
 
-    // For <annotation-xml>, we follow these definitions from the HTML5 RelaxNG schema:
-    // - annotation-xml.model.mathml
-    // - annotation-xml.model.svg
-    // - annotation-xml.model.xhtml
-
     ASSERT(hasTagName(annotation_xmlTag));
-    auto& value = attributeWithoutSynchronization(encodingAttr);
-
-    if (is<MathMLElement>(child) && (MathMLSelectElement::isMathMLEncoding(value) || MathMLSelectElement::isHTMLEncoding(value))) {
-        auto& mathmlElement = downcast<MathMLElement>(child);
-        return is<MathMLMathElement>(mathmlElement);
-    }
-
-    if (is<SVGElement>(child) && (MathMLSelectElement::isSVGEncoding(value) || MathMLSelectElement::isHTMLEncoding(value))) {
-        auto& svgElement = downcast<SVGElement>(child);
-        return is<SVGSVGElement>(svgElement);
-    }
-
-    if (is<HTMLElement>(child) && MathMLSelectElement::isHTMLEncoding(value)) {
-        auto& htmlElement = downcast<HTMLElement>(child);
-        return is<HTMLHtmlElement>(htmlElement) || (isFlowContent(htmlElement) && StyledElement::childShouldCreateRenderer(child));
-    }
-
-    return false;
```

Comment:
Can we just defer to the parent class of MathMLElement:

`return StyledElement::childShouldCreateRenderer(child);`

? Checking a bit, this will call Element::childShouldCreateRenderer where the difference is minimal (for SVG child only accept valid `<svg>`).